### DISCUSSION TITLE
ecs-cluster service catalog ebs volume delete on

### DISCUSSION THREAD

**Discussion by swb-buildarchetype** [gruntwork-io/knowledge-base#799] [How do I] [answered]
How can I get ecs cluster service catalog modules to allow EBS volumes to delete on termination? Racking up huge bills every time a pipeline runs in dev and hoping to have dev instances default to EBS volume destroying on detach. This is available as a variable on the ecs-cluster module itself: https://github.com/gruntwork-io/terraform-aws-ecs/blob/7c6fbc9731903ce175703608ac8d429c01201604/modules/ecs-cluster/main.tf#L135

It, however, does not appear to be available on the service catalog wrapper for the module. Can we please expose this? Is there a way to pass these sort of custom variables along to the ecs-cluster? I am thinking no, and that it needs to be exposed as a variable.

---

<ins datetime="2024-01-20T19:44:13Z">
  <p><a href="https://support.gruntwork.io/hc/requests/110709">Tracked in ticket #110709</a></p>
</ins>

**Comment by gcagle3** [comment]
Hi @swbmindcycle, thank you for submitting this question! To clarify, can you confirm that you're referring to the service catalog module 'modules/services/ecs-service' specifically for this request? If so, we can create a quick issue to track this and should be able to turn around a fix fairly quickly. Thank you!

**Reply by gcagle3** [reply, marked as answer]
@swbmindcycle the EBS delete on terminate for the ecs-cluster service can now be toggled using the var `cluster_instance_ebs_delete_on_termination` in version [v0.108.3](https://github.com/gruntwork-io/terraform-aws-service-catalog/releases/tag/v0.108.3). Please test this out at your convenience.

**Comment by swb-buildarchetype** [comment]
Confirmed. That is what I’m referring to. While you’re at it I also opened a pr to default to gp3 for all ebs volumes attached to ECS clusters in this same module. Would love to default to gp3 as this is current AWS standard and then also have a flag to pass it through to the module via terragrunt.hcl configIn general it would be great to extend all properties of the modules to the service catalog but just make them optional. Is there a way to do this now or is it still hard coded?—📱 sent from my mobile deviceSam Ward BiddleCEO and Principal ***@***.***“a computer is like a bicycle for the mind”On Jan 23, 2024, at 4:41 PM, gcagle3 ***@***.***> wrote:﻿
Hi @swbmindcycle, thank you for submitting this question! To clarify, can you confirm that you're referring to the service catalog module 'modules/services/ecs-service' specifically for this request? If so, we can create a quick issue to track this and should be able to turn around a fix fairly quickly. Thank you!

**Reply by gcagle3** [reply]
Fantastic, thank you for confirming. Let me take a look at the PR for the ECS cluster module, then we can update the service catalog to pull in that change and expose the volume vars.

**Reply by gcagle3** [reply]
Your PR in the ECS cluster module has been merged and is included in release [v0.35.14](https://github.com/gruntwork-io/terraform-aws-ecs/releases/tag/v0.35.14). We will follow up with an update to the service catalog next to expose the EBS volume delete option. Thank you for your patience!

**Reply by gcagle3** [reply]
This has also been made available in our service catalog as part of release [v0.108.6](https://github.com/gruntwork-io/terraform-aws-service-catalog/releases/tag/v0.108.6).

**Comment by swb-buildarchetype** [comment]
Amazing! This is really helpful. Thanks—📱 sent from my mobile deviceSam Ward BiddleCEO and Principal ***@***.***“a computer is like a bicycle for the mind”On Feb 9, 2024, at 2:29 PM, gcagle3 ***@***.***> wrote:﻿
This has also been made available in our service catalog as part of release v0.108.6.
